### PR TITLE
Sanitize console input

### PR DESCRIPTION
**Pull request by FreddieBanfield**
Introduced getInput: A function that gets the users input. It takes in a few different parameters to allow the developer to specify what kind of sanitization needs to happen on this input

Introduced checkInputType: requires the input_type parameter. Checks if the user inputted an int, float or str

Introduced restrictInputToRangeInclusive: Requires min and max parameters. Checks whether a number is within a range of numbers. Good for getting days (1-31), month (1-12), years

Fixed name issue with CALANDER

Fixed name issue with Cibbous

Added tests

### CODE REVIEW

File: AstroCal/view/console_output.py [hunk 1]
```diff
@@ -59,51 +68,55 @@ def sun_menu():
...
 # moon options
 def moon_menu():
+    clear()
+    print('✧ ･ﾟ * ✧  ASTRO CALENDAR  ✧ ･ﾟ * ✧ ･ﾟ \n')
     print('Moon Events')
     print('1. View Today')
     print('2. Lunar Eclipse')
     print('3. View Moon Status')
     print('4. Date of Next New Moon')
     print('5. Date of Next Full Moon \n')
```
Comment (nmiles01) [inline AstroCal/view/console_output.py:117]: we have an extra space not sure if intended but not consistent with sun

**Review by nmiles01 — COMMENTED** [review]
Not sure if fixing the spacing on the menu is important in this branch but I did notice it.
Also does not break code but can have any amount of leading zeros and still use the optinon
ig( 0000000000000000000002 will give option 2

File: AstroCal/view/console_output.py [hunk 2]
```diff
@@ -1,43 +1,52 @@
 # Astro Calender Console Menu
 
 # Imports
+from asyncio.windows_events import NULL
```
Comment (jakeljns) [inline AstroCal/view/console_output.py]: This import does not work on MacOS

**Review by jakeljns — DISMISSED** [review]
Please remove windows specific import for NULL as we should not be programming for Windows, and instead be platform agnostic so that it will work on macOS (the primary platform)

File: AstroCal/view/console_output.py [hunk 3]
```diff
@@ -3,41 +3,49 @@
...
-from control import control
+from pickle import TRUE
+from AstroCal.control import control
 import sys
+import os
 
 
 # Main Menu
 def main_menu():
-    print('✧ ･ﾟ * ✧  ASTRO CALANDER  ✧ ･ﾟ * ✧ ･ﾟ \n')
```
Comment (MellowYeller) [inline AstroCal/view/console_output.py:12]: I think we should revert and trademark CALANDER

Comment (FreddieBanfield) [reply AstroCal/view/console_output.py:12]: Done

File: AstroCal/view/console_output.py [hunk 4]
```diff
@@ -12,7 +12,7 @@
 # Main Menu
 def main_menu():
     clear()
-    print('✧ ･ﾟ * ✧  ASTRO CALENDAR  ✧ ･ﾟ * ✧ ･ﾟ \n')
+    print('✧ ･ﾟ * ✧  ASTRO CALANDER  ✧ ･ﾟ * ✧ ･ﾟ \n')
```
Comment (MellowYeller) [inline AstroCal/view/console_output.py:15]: It's perfect 🤩

**Review by MellowYeller — APPROVED** [review]

**Review by nmiles01 — APPROVED** [review]

**Review by jakeljns — APPROVED** [review]

**Pull request merged**